### PR TITLE
csv discovery takes too much memory (#2089)

### LINKED ISSUE #2089: csv discovery takes too much memory
## Expected Behavior
Discovery should work with memory requirements scaling based on number of columns, not the number of rows.

## Current Behavior
On a t2.large instance, discovering a 5gb CSV file freezes and gets killed as OOM.

## Severity of the bug for you
High

We probably need to mess with how our loaded file is stored in memory or change the type inference process to scan rows without storing them all in memory.

### PR DESCRIPTION
**Pull request by Zirochkaa**
Use `chunksize` parameter when reading csv files.

## What
This PR is for https://github.com/airbytehq/airbyte/issues/2089 issue.

## How
*Describe the solution*
Use `chunksize` parameter when reading csv files. It'll load not all the file but its chunks one at a time. Each chunk has 10000 lines.

## Pre-merge Checklist
- [x] *Run integration tests*
- [x] *Publish Docker images*

## Recommended reading order
1. `airbyte-integrations/connectors/source-file/source_file/client.py`

### CODE REVIEW

File: airbyte-integrations/connectors/source-file/source_file/client.py [hunk 1]
```diff
@@ -322,10 +324,17 @@ def read(self, fields: Iterable = None) -> Iterable[dict]:
             if self._reader_format == "json" or self._reader_format == "jsonl":
                 yield from self.load_nested_json(fp)
             else:
+                fields = list(fields)
                 for df in self.load_dataframes(fp):
-                    columns = set(fields).intersection(set(df.columns)) if fields else df.columns
-                    df = df.replace(np.nan, "NaN", regex=True)
-                    yield from df[columns].to_dict(orient="records")
+                    if self._reader_format == "csv" and isinstance(df, pd.io.parsers.TextFileReader):
```
Comment (keu) [inline airbyte-integrations/connectors/source-file/source_file/client.py:329]:
```python
chunks = []
if self._reader_format == "csv" and isinstance(df, pd.io.parsers.TextFileReader):
    chunks = df
```

File: airbyte-integrations/connectors/source-file/source_file/client.py [hunk 2]
```diff
@@ -286,9 +286,11 @@ def load_dataframes(self, fp, skip_data=False) -> List:
             raise ConfigurationError(error_msg) from err
 
         reader_options = {**self._reader_options}
-        if skip_data and self._reader_format == "csv":
-            reader_options["nrows"] = 0
-            reader_options["index_col"] = 0
+        if self._reader_format == "csv":
```
Comment (keu) [inline airbyte-integrations/connectors/source-file/source_file/client.py:289]: lets maybe return Iterable and then
```python
if self._reader_format == "csv":
 ....
    yield from reader(fp, **reader_options)
else:
    yield reader(fp, **reader_options)
```

**Review by keu — CHANGES_REQUESTED** [review]
see my comments

File: airbyte-integrations/connectors/source-file/source_file/client.py (hunk 1 above)
Comment (keu) [reply airbyte-integrations/connectors/source-file/source_file/client.py:329]: if next suggestion applied then this one don't needed

Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-file/source_file/client.py:331]: For now, I implemented this version. Let's wait for @vitaliizazmic's review also.

Comment (vitaliizazmic) [reply airbyte-integrations/connectors/source-file/source_file/client.py:331]: I think, it's better to change load_dataframes method

**Review by vitaliizazmic — COMMENTED** [review]
load_dataframes method requires changes according to @keu comment

File: airbyte-integrations/connectors/source-file/source_file/client.py (hunk 1 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-file/source_file/client.py:331]: Ok.

File: airbyte-integrations/connectors/source-file/source_file/client.py [hunk 3]
```diff
@@ -286,11 +286,15 @@ def load_dataframes(self, fp, skip_data=False) -> List:
...
+            reader_options["chunksize"] = 10000
+            if skip_data:
+                reader_options["nrows"] = 0
+                reader_options["index_col"] = 0
 
-        return [reader(fp, **reader_options)]
+            yield from reader(fp, **reader_options)
+        else:
+            yield reader(fp, **reader_options)
```
Comment (keu) [inline airbyte-integrations/connectors/source-file/source_file/client.py:297]: you need to update return annotations -> `Iterable`

**Review by keu — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-file/source_file/client.py (hunk 3 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-file/source_file/client.py:297]: Done.

**Review by keu — APPROVED** [review]

**Review by vitaliizazmic — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]

**Pull request merged**